drivers/adc/adc_lmp90xxx.c

```diff
@@ -0,0 +1,1136 @@
+		err = lmp90xxx_write_reg8(dev, LMP90XXX_REG_SPI_CRC_CN,
+					LMP90XXX_EN_CRC(1) |
+					LMP90XXX_DRDYB_AFT_CRC(1));
+		if (err) {
+			LOG_ERR("failed to enable CRC (err %d)", err);
+			return err;
+		}
+	}
+
+	if (LMP90XXX_HAS_DRDYB(config)) {
+		drdyb_dev = device_get_binding(config->drdyb_dev_name);
+		if (!drdyb_dev) {
+			LOG_ERR("DRDYB GPIO device '%s' not found",
+				config->drdyb_dev_name);
+			return -EINVAL;
+		}
+
+		err = gpio_pin_configure(drdyb_dev, config->drdyb_pin,
+					 (GPIO_DIR_IN | GPIO_INT |
+					 GPIO_INT_EDGE | GPIO_INT_ACTIVE_LOW));
+		if (err) {
+			LOG_ERR("failed to configure DRDYB GPIO pin (err %d)",
+				err);
+			return -EINVAL;
+		}
+
+		gpio_init_callback(&data->drdyb_cb, lmp90xxx_drdyb_callback,
+				   BIT(config->drdyb_pin));
+
+		err = gpio_add_callback(drdyb_dev, &data->drdyb_cb);
+		if (err) {
+			LOG_ERR("failed to add DRDYB callback (err %d)", err);
+			return -EINVAL;
+		}
+
+		err = lmp90xxx_write_reg8(dev, LMP90XXX_REG_SPI_DRDYBCN,
+					  LMP90XXX_SPI_DRDYB_D6(1));
+		if (err) {
+			LOG_ERR("failed to configure D6 as DRDYB (err %d)",
+				err);
+			return err;
+		}
+
+		err = gpio_pin_enable_callback(drdyb_dev,
+					       config->drdyb_pin);
+		if (err) {
+			LOG_ERR("failed to enable DRDBY callback (err %d)",
+				err);
+			return -EINVAL;
+		}
+	}
+
+	k_thread_create(&data->thread, data->stack,
+			CONFIG_ADC_LMP90XXX_ACQUISITION_THREAD_STACK_SIZE,
+			(k_thread_entry_t)lmp90xxx_acquisition_thread,
+			dev, NULL, NULL,
+			CONFIG_ADC_LMP90XXX_ACQUISITION_THREAD_PRIO,
+			0, K_NO_WAIT);
+
+	/* Put device in stand-by to prepare it for single-shot conversion */
+	err = lmp90xxx_write_reg8(dev, LMP90XXX_REG_PWRCN, LMP90XXX_PWRCN(0x3));
+	if (err) {
+		LOG_ERR("failed to request stand-by mode (err %d)", err);
+		return err;
+	}
+
+	adc_context_unlock_unconditionally(&data->ctx);
+
+	return 0;
+}
+
+static const struct adc_driver_api lmp90xxx_adc_api = {
+	.channel_setup = lmp90xxx_adc_channel_setup,
+	.read = lmp90xxx_adc_read,
+#ifdef CONFIG_ADC_ASYNC
+	.read_async = lmp90xxx_adc_read_async,
+#endif
+};
+
+#define ASSERT_LMP90XXX_CURRENT_VALID(v) \
+	BUILD_ASSERT_MSG(v == 0 || v == 100 || v == 200 || v == 300 || \
+			 v == 400 || v == 500 || v == 600 || v == 700 || \
+			 v == 800 || v == 900 || v == 1000, \
+			 "unsupported RTD current (" #v ")")
+
+#define LMP90XXX_UAMPS_TO_RTD_CUR_SEL(x) (x / 100)
+
+#define LMP90XXX_DEVICE(t, n, res, ch) \
+	ASSERT_LMP90XXX_CURRENT_VALID(DT_INST_##n##_TI_LMP##t##_RTD_CURRENT); \
+	static const struct lmp90xxx_config lmp##t##_config_##n = { \
+		.spi_dev_name = DT_INST_##n##_TI_LMP##t##_BUS_NAME, \
+		.spi_cs_dev_name = \
+			DT_INST_##n##_TI_LMP##t##_CS_GPIOS_CONTROLLER, \
+		.spi_cs_pin = DT_INST_##n##_TI_LMP##t##_CS_GPIOS_PIN, \
+		.drdyb_dev_name = \
+			DT_INST_##n##_TI_LMP##t##_DRDYB_GPIOS_CONTROLLER, \
+		.drdyb_pin = DT_INST_##n##_TI_LMP##t##_DRDYB_GPIOS_PIN, \
+		.drdyb_flags = DT_INST_##n##_TI_LMP##t##_DRDYB_GPIOS_FLAGS, \
```

Comment:
Is this field intended to be used in some future or should its value be taken now instead of `GPIO_INT_ACTIVE_LOW` when the DRDYB pin is configured?

Reply:
@anangl It was intended to be used for the pin configuration. Fixed.